### PR TITLE
Add persist_mut and persist_mut_keyed for non-monitone deletions

### PR DESCRIPTION
**Pull request by zzlk**

### CODE REVIEW

**Comment by MingweiSamuel** [comment]
We should pick a name other than `'mut`? Any ideas?

**Comment by zzlk** [comment]
> We should pick a name other than `'mut`? Any ideas?

we decided to change this to mutable

**Review by MingweiSamuel — APPROVED** [review]

File: hydroflow/src/util/sparse_vec.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    pub fn delete(&mut self, item: &T) {
+        if let Some(indices) = self.hash_map.get(item) {
+            for index in indices {
+                self.vec[*index] = None;
+            }
+        }
+    }
+
+    /// Iterate through all items in the vector in order. Deleted items will not appear in the iteration.
+    pub fn iter(&self) -> impl Iterator<Item = &T> {
```
[the remark this answers is not in the document]
Comment (zzlk) [reply hydroflow/src/util/sparse_vec.rs:32]: Debug requires T: Debug, but the others are ok.

File: hydroflow_lang/src/graph/ops/persist_mut.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                   ..
+               },
+               diagnostics| {
+        assert!(is_pull);
+
+        if matches!(persistence_args[..], [Persistence::Tick]) {
+            diagnostics.push(Diagnostic::spanned(
+                op_span,
+                Level::Error,
+                format!("`{}()` can only have `'static` persistence.", op_name),
```
[the remark this answers is not in the document]
Comment (zzlk) [reply hydroflow_lang/src/graph/ops/persist_mut.rs:64]: Changed them all to RANGE_0 persistence args

File: hydroflow/tests/surface_codegen.rs [hunk 3]
```diff
@@ -807,3 +807,90 @@ pub fn test_covid_tracing() {
...
+
+    assert_eq!(
+        HashSet::from_iter([1, 3, 4]),
+        collect_ready::<HashSet<_>, _>(&mut pull_rx)
+    );
+    assert_eq!(
+        HashSet::from_iter([1, 4]),
+        collect_ready::<HashSet<_>, _>(&mut push_rx)
+    );
+}
```
[the remark this answers is not in the document]
Comment (zzlk) [reply hydroflow/tests/surface_codegen.rs:896]: Moved them to surface_persist.rs

**Pull request merged**